### PR TITLE
BUG: astype(Int64) raises AttributeError

### LINKED ISSUE #22578: BUG: astype('Int64') raises AttributeError
```
In [8]: df = pd.DataFrame([[1., 2.], [3., 4.], [5., 6.]], columns=['a', 'b'])

In [9]: df
Out[9]: 
     a    b
0  1.0  2.0
1  3.0  4.0
2  5.0  6.0

In [10]: df.astype('Int64')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-10-b9d2763e69d8> in <module>()
----> 1 df.astype('Int64')

~/scipy/pandas/pandas/util/_decorators.py in wrapper(*args, **kwargs)
    175                 else:
    176                     kwargs[new_arg_name] = new_arg_value
--> 177             return func(*args, **kwargs)
    178         return wrapper
    179     return _deprecate_kwarg

~/scipy/pandas/pandas/core/generic.py in astype(self, dtype, copy, errors, **kwargs)
   5162             # else, only a single dtype is given
   5163             new_data = self._data.astype(dtype=dtype, copy=copy, errors=errors,
-> 5164                                          **kwargs)
   5165             return self._constructor(new_data).__finalize__(self)
   5166 

~/scipy/pandas/pandas/core/internals/managers.py in astype(self, dtype, **kwargs)
    554 
    555     def astype(self, dtype, **kwargs):
--> 556         return self.apply('astype', dtype=dtype, **kwargs)
    557 
    558     def convert(self, **kwargs):

~/scipy/pandas/pandas/core/internals/managers.py in apply(self, f, axes, filter, do_integrity_check, consolidate, **kwargs)
    421 
    422             kwargs['mgr'] = self
--> 423             applied = getattr(b, f)(**kwargs)
    424             result_blocks = _extend_blocks(applied, result_blocks)
    425 

~/scipy/pandas/pandas/core/internals/blocks.py in astype(self, dtype, copy, errors, values, **kwargs)
    562     def astype(self, dtype, copy=False, errors='raise', values=None, **kwargs):
    563         return self._astype(dtype, copy=copy, errors=errors, values=values,
--> 564                             **kwargs)
    565 
    566     def _astype(self, dtype, copy=False, errors='raise', values=None,

~/scipy/pandas/pandas/core/internals/blocks.py in _astype(self, dtype, copy, errors, values, klass, mgr, **kwargs)
    679                     "current ({newb_dtype} [{newb_size}])".format(
    680                         copy=copy, dtype=self.dtype.name,
--> 681                         itemsize=self.itemsize, newb_dtype=newb.dtype.name,
    682                         newb_size=newb.itemsize))
    683         return newb

AttributeError: 'FloatBlock' object has no attribute 'itemsize'
```

The same if the starting dataframe is of integer dtype.

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] closes #22578
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -671,15 +671,6 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
                 raise
             newb = self.copy() if copy else self
 
-        if newb.is_numeric and self.is_numeric:
-            if newb.shape != self.shape:
```
Comment (sinhrks) [inline pandas/core/internals/blocks.py]: Does anyone know what case meets the condition?

Comment (jreback) [reply pandas/core/internals/blocks.py]: I *think* .astype('i1') on an int64 might hit this

Comment (sinhrks) [reply pandas/core/internals/blocks.py]: Thx. It looks work (might depending on OS?)

```
pd.DataFrame([1]).astype('i1')
```

To keep the condition for safety, remove itemsize from the error message? (or add `.itemsize` to extension dtypes)

```
pd.Series([1], dtype='int64').dtype.itemsize
# 8
pd.Series([1], dtype='Int64').dtype.itemsize
# AttributeError: 'Int64Dtype' object has no attribute 'itemsize'
```

**Comment by jreback** [comment]
looks good. can you add on the issue to the whatsnew for integer array (just add to the list), and rebase.

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -675,11 +675,11 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
             if newb.shape != self.shape:
                 raise TypeError(
                     "cannot set astype for copy = [{copy}] for dtype "
-                    "({dtype} [{itemsize}]) with smaller itemsize than "
-                    "current ({newb_dtype} [{newb_size}])".format(
+                    "({dtype} [{shape}]) to different shape "
```
Comment (sinhrks) [inline pandas/core/internals/blocks.py]: Changed the message not to raise `AttributeError`.

**Comment by sinhrks** [comment]
Thx for the review. Updated and rebased.

**Comment by jreback** [comment]
seems there is a lint error.

**Comment by sinhrks** [comment]
Sorry, fixed.

**Comment by jreback** [comment]
thanks!

**Pull request merged**